lib/mayaUsd/render/pxrUsdMayaGL/shapeAdapter.h

```diff
@@ -152,52 +155,87 @@ class PxrMayaHdShapeAdapter
         /// If there is no corresponding HdReprSelector for the given display
-        /// state, an empty HdReprSelector is returned.
+        /// style, an empty HdReprSelector is returned.
         MAYAUSD_CORE_PUBLIC
-        virtual HdReprSelector GetReprSelectorForDisplayState(
-                const unsigned int displayStyle,
-                const MHWRender::DisplayStatus displayStatus) const;
+        HdReprSelector GetReprSelectorForDisplayStyle(
+                unsigned int displayStyle) const;
+
+        /// Get the render params for the shape adapter's current state.
+        const PxrMayaHdRenderParams& GetRenderParams() const {
+            return _renderParams;
+        }
 
-        /// Get a set of render params from the shape adapter's current state.
+        /// Get the rprim collection for the given repr.
         ///
-        /// Sets \p drawShape and \p drawBoundingBox depending on whether shape
-        /// and/or bounding box rendering is indicated from the state.
-        MAYAUSD_CORE_PUBLIC
-        virtual PxrMayaHdRenderParams GetRenderParams(
-                bool* drawShape,
-                bool* drawBoundingBox) const;
+        /// These collections are created when the shape adapter's MDagPath is
+        /// set to a valid Maya shape.
+        ///
+        /// Returns an empty collection if there is no collection for the given
+        /// repr.
+        const HdRprimCollection& GetRprimCollection(
+                const HdReprSelector& repr) const {
+            const auto& iter = _rprimCollectionMap.find(repr);
+            if (iter != _rprimCollectionMap.cend()) {
+                return iter->second;
+            }
+
+            static const HdRprimCollection emptyCollection;
+            return emptyCollection;
+        }
 
-        MAYAUSD_CORE_PUBLIC
-        virtual const HdRprimCollection& GetRprimCollection() const;
-
-        /// Retrieves the render tags for this shape.  I.e. which
-        /// prim purposes should be drawn (such as geomerty, proxy, guides
-        /// and/or render).
-        /// This function just returns the _renderTags attribute and it
-        /// is expected each subclass update the attribute in _Sync() or
-        /// overrides this function if it needs special processing.
-        MAYAUSD_CORE_PUBLIC
-        virtual const TfTokenVector& GetRenderTags() const;
+        /// Get the ID of the render task for the collection of the given repr.
+        ///
+        /// These render task IDs are created when the shape adapter's MDagPath
+        /// is set to a valid Maya shape.
+        ///
+        /// Returns an empty SdfPath if there is no render task ID for the
+        /// given repr.
+        const SdfPath& GetRenderTaskId(const HdReprSelector& repr) const {
+            const auto& iter = _renderTaskIdMap.find(repr);
+            if (iter != _renderTaskIdMap.cend()) {
+                return iter->second;
+            }
+
+            static const SdfPath emptyTaskId;
```

Comment:
This was primarily a performance consideration to minimize copying, since `GetRenderTaskId()` is going to get called once per proxy shape for every frame draw.

Though now that I'm looking at this, I think I could have used `return SdfPath::EmptyPath()` here. :/